### DISCUSSION TITLE
Getting the Raw Query?

### DISCUSSION THREAD

**Discussion by mexicantexan** [goadesign/goa#3511] [Q&A]
How do I get access to the raw query? Is it in the context? If so, what would the key be? Or is there a way to do this in the design file? The index/order of the parameters I'm trying to get from the raw query is important, and can happen in any order so I don't think parsing them in the design file is the correct approach since I believe they'll lose their original order.

Right now I have this:
```go
httpRequest, _ := ctx.Value({{NOT_SURE_WHAT_TO_PUT_HERE}}).(*http.Request)
rawQuery := httpRequest.URL.RawQuery
```
which I gathered from the following two pieces of documentation:

- https://goa.design/reference/goa/#func-contextresponse-a-name-goa-responsedata-contextresponse-a
- https://goa.design/implement/context/

Is this the right way to do this?

**Comment by raphael** [comment]
Hello, the [PopulateRequestContext](https://github.com/goadesign/goa/blob/v3/http/middleware/request.go) middleware inserts  HTTP request fields into the request context including the raw request URI which contains the query string. Here is an example:

```go
// ...
calcEndpoints = calc.NewEndpoints(calcSvc)
mux = goahttp.NewMuxer()
calcServer = calcsvr.New(calcEndpoints, mux, goahttp.RequestDecoder, goahttp.ResponseEncoder, nil, nil, nil)
calcSever.Use(middleware.PopulateRequestContext())
// ...
```

The URI can then be retrieved in the method using:

```go
uri := ctx.Value(middleware.RequestURIKey)
```

Alternatively you could write your own HTTP middleware that populates the parsed query string.

Hope this helps!
